### ISSUE TITLE
Switch between closed line and area

### ISSUE THREAD

**Issue by zenfiric** [openstreetmap/iD#1828]
Sometimes I mistakenly chose line when I want to draw an area. There is no way to quickly switch between these two. I could delete a line and then draw an area or I could chose Other from a preset list and then add and edit individual line.
So, radial menu should have another option like "Make it an area/closed line" or when I click on this line an then on a button "Area" it should automatically  tag it with an area tag.

**Comment by sulfo** [comment]
+1. Really need this for fixing areas there should be lines. If I cut the area, then all the data is gone for the path, so it would be nice with his feature.

**Comment by homersimpsons** [comment]
Would love this too to solve "multipolygon is a polygon" (issue on osmose) easily.

**Comment by bhousel** [comment]
> Would love this too to solve "multipolygon is a polygon" (issue on osmose) easily.

FYI, if you're a power user cleaning up osmose issues, you can solve it easily by adding the `area=yes` tag in the raw tag editor - this will make all the area presets available.

But I agree, we still do want to add this feature for non-power users.

**Comment by homersimpsons** [comment]
Yes sorry, I'm far away to be a pro at OSM, I try to map my city and around, and I prefer far away OSM because it's really fast (compare to JOSM harder to use) and beautiful (compare to Potlatch 2)

**Comment by bhousel** [comment]
For fun I tried to build this today:
```js
{
    "icon": "multipolygon",
    "name": "Convert To Area",
    "tags": {"area": "yes"},
    "geometry": ["line"],
    "matchScore": 0.0
}
```
```js
{
    "icon": "other-line",
    "name": "Convert To Line",
    "tags": {"area": "no"},
    "geometry": ["area"],
    "matchScore": 0.0
}
```

And it works ok, but the problem is you can use it to convert non-closed lines into areas and make some pretty broken geometries that way.   "Convert to Area" should really be allowed only on linear rings.

**Comment by 1ec5** [comment]
I often experience a need for this functionality when refining a row of conjoined buildings that was mapped as a single building. I prefer to split up the existing way instead of redrawing the entire building, which can often be tedious. My workaround:

1. Change `building=yes` to `buildingX=yes`, causing the area to become a way.
1. Split up the way at the points where each building meets another building. (iD makes this easy by positioning the endpoints of the resulting ways at all the right places.)
1. Close the ways.
1. Change `buildingX=yes` back to `building=yes`.

**Comment by jidanni** [comment]
Indeed, the user may have started by clicking "area", but after hours of tracing the edge and clicking various presets etc. etc., he might find he is now dealing with a line, and panic.
Without knowing he can recover by simply adding area=yes, he might, dejected, think he has no other choice than retracing the entire geometry. So it would be nice for him to be able to find a "toggle area/ line" button.

**Comment by aaronaxvig** [comment]
Today I stumbled upon a way to convert a closed line to an area.

1. With zero tags on the line, type `area `as the name of a new tag.
1. Tab out of the tag name box, and observe that the Feature Type at the top of the edit pane has changed to Area.
1. Click Area, and you are able to change it to any area-type Feature Type that you desire (building, parking lot, etc.).
1. After choosing the Feature Type, the tag with the name `area `is not visible any more.

**Comment by michaelblyons** [comment]
@aaronaxvig If you toggle the tags to the raw view with this UI:

<img width="66" alt="Screen Shot 2021-03-17 at 3 44 37 PM" src="https://user-images.githubusercontent.com/2701605/111528557-37075a80-8759-11eb-84b4-74dd967264c4.png">

and then type `area=yes`, you can toggle it back and pick your feature type without clicking out and back. I believe this is what @bhousel was saying [up here](https://github.com/openstreetmap/iD/issues/1828#issuecomment-219835881).

**Comment by greatvovan** [comment]
Tagging with `area=yes` is easy and fast, however it is not at all obvious for beginner users. Need a button in context menu to do that.

**Comment by sur0g** [comment]
I suggest a new button in the context menu (right mouse). That was the first place a thought it would be. After realizing there's no such button, I had to search for the solution (which lead me to this issue). It's counterintuitive.
In addition, iD is a standalone editor therefore it must implement its own interface to represent and manipulate the current XML map format.

**Comment by frspp** [comment]
+1 . Frustrating.
